PSReadLine.build.ps1

```diff
@@ -202,6 +202,9 @@ task LayoutModule BuildMainModule, BuildMamlHelp, {
 
     Copy-Item PSReadLine/bin/$Configuration/Microsoft.PowerShell.PSReadLine2.dll $targetDir
     Copy-Item PSReadLine/bin/$Configuration/System.Runtime.InteropServices.RuntimeInformation.dll $targetDir
+    Copy-Item PSReadLine/bin/$Configuration/UnixConsoleEcho.dll $targetDir
+    Copy-Item PSReadLine/bin/$Configuration/libdisablekeyecho.so $targetDir
```

Comment:
Presumably this is an x64 binary? What about x86 or arm32 or arm64?

I see that the code comes from the CLR. Is there any reasonable way to ask the CLR to expose an API instead?

Reply:
For x86, it looks [it isn't supported](https://github.com/dotnet/core/blob/master/release-notes/2.0/2.0-supported-os.md) on Linux and Mac.  I do need to look into what I need to do for arm* though.

> Is there any reasonable way to ask the CLR to expose an API instead?

That definitely would have been preferred.  [Here's the thread](https://github.com/dotnet/corefx/issues/16352) we asked about calling their library for disabling input echo.  That's where they suggested stripping out their implementation.  And [here's](https://github.com/dotnet/corefx/issues/25036) the request to create an async version of `ReadKey`.

Reply:
Can you just run `stty -echo` instead? If you do that once before the call to PSReadLine and once to restore after, it seems not too bad. See https://stackoverflow.com/questions/5633472/how-do-i-turn-off-echo-in-a-terminal.

Reply:
I'm definitely open to the idea and looking into it.  Just to clarify though, when you say the call to PSReadLine do you mean call it from inside the `ReadLine` method or only call it from editor services?  Disabling input echo is needed to implement `WaitForKeyAvailable`.  If it was only disabled by the editor services host, every key press would appear to be rendered twice when in the standard host.

Reply:
Yeah, I was referring to the method `ReadLine`. On Windows, we save the console mode at the beginning and restore it at the end, that's similar to what I was proposing for `stty`.

Reply:
Any reason why this needs to be supplied as an arch-dependent binary? Why not use P/Invokes like is done for Windows with Get/SetConsoleMode?

Reply:
@parkovski ...well I wish I had a good explanation for why I originally decided it couldn't be done with pure P/Invokes.  But you're 100% correct, it definitely can.  I just wrote up something basic to test and it worked great.

Only needed to import `tcsetattr` and `tcgetattr` from libc. They should be available everywhere, yeah? Thank you, can't believe I over looked that.  I'll get started on switching to P/Invokes.